spec/controllers/mixins/actions/host_actions/misc_spec.rb

```diff
@@ -0,0 +1,24 @@
+describe Mixins::Actions::HostActions::Misc do
+  describe "#host_button_operation" do
+    let(:controller) do
+      return HostController.new
+    end
+    let(:admin_user) { FactoryGirl.create(:user, :role => "super_administrator") }
+    let!(:host) { FactoryGirl.create(:host) }
+
+    before do
+      allow(controller).to receive(:find_checked_items).and_return(Host.all)
+      stub_user(:features => :all)
+      EvmSpecHelper.create_guid_miq_server_zone
+      login_as admin_user
+      allow(User).to receive(:current_user).and_return(admin_user)
+      allow(controller).to receive(:process_hosts).with([host.id], 'refresh_ems', 'Refresh')
+      controller.instance_variable_set(:@_params, {})
```

Comment:
You can set the instance variable `@params` correctly
`controller.instance_variable_set(:@_params, :id=> Host.all.id)`
and remove 
`allow(controller).to receive(:find_checked_items).and_return(Host.all)`